admin/eliotlogging.py

```diff
@@ -0,0 +1,51 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Filter Eliot logs out of Twisted test.log.
+"""
+
+import sys
+
+from flocker.testtools._base import _get_eliot_data
+
+
+def chain_files(paths):
+    """
+    Iterate through many files, line by line.
+    """
+    for path in paths:
+        with open(path, 'r') as f:
+            for line in f:
+                yield line
+
+
+def process_line(line):
+    eliot_line = _get_eliot_data(line)
```

Comment:
:grey_question: I think _get_eliot_data can raise ValueErrors if the line does not have the ELIOT marker in it. It'd be nice if a single bad line in the logs didn't cause the whole utility to crash (My biggest grievance about eliot-tree and the like).

The tool I want the most would be one that would print to STDERR the line numbers that had errors and what the errors were (perhaps aggregating similar errors), and would print out all lines that didn't have errors on stdout for consumption by eliot-tree or whatnot.